backend/src/ee/helpers/secret.ts

```diff
@@ -45,15 +73,23 @@ const takeSecretSnapshotHelper = async ({
       },
     ]).exec()
   ).map((s) => s.versionId);
+  const latestFolderVersion = await FolderVersion.findOne({
+    environment,
+    workspace: workspaceId,
+    "nodes.id": folderId || "root",
+  }).sort({ "nodes.version": -1 });
```

Comment:
What do you think about adding a index for this query so that data is always ordered on that index?

Reply:
Something like:

```
FolderVersionSchema.index({
  environment: 1,
  workspace: 1,
  "nodes.id": 1,
  "nodes.version": -1
});
```

Reply:
Yah indexing would make this faster